service/src/NVMLDevicePool.cpp

```diff
@@ -150,6 +150,30 @@ namespace geopm
         return (uint64_t)result;
     }
 
+    std::vector<unsigned int> NVMLDevicePoolImp::frequency_supported_sm(int accel_idx) const
+    {
+        check_accel_range(accel_idx);
+        unsigned int count = 100;
+        std::vector<unsigned int> supported_freqs(count);
+
+        nvmlReturn_t nvml_result = nvmlDeviceGetSupportedGraphicsClocks(m_nvml_device.at(accel_idx),
+                                                                        frequency_status_mem(accel_idx),
+                                                                        &count, supported_freqs.data());
+
+        if (nvml_result == NVML_ERROR_INSUFFICIENT_SIZE) {
+            supported_freqs.resize(count);
+            nvml_result = nvmlDeviceGetSupportedGraphicsClocks(m_nvml_device.at(accel_idx),
+                                                               frequency_status_mem(accel_idx),
+                                                               &count, supported_freqs.data());
+        }
+        check_nvml_result(nvml_result, GEOPM_ERROR_RUNTIME, "NVMLDevicePool::" + std::string(__func__) +
+                          ": NVML failed to get SM Frequency for accelerator " +
+                          std::to_string(accel_idx) + ".", __LINE__);
+
+        return supported_freqs;
```

Comment:
don't we want to
```
supported_freqs.resize(count);
```
before returning a length 100 vector?

Reply:
Re-arrange the code a bit so only a single resize should be needed.